src/PlaywrightSharp/Page.cs

```diff
@@ -357,6 +361,47 @@ public Task<IPage> GetOpenerAsync()
 
         /// <inheritdoc cref="IPage.WaitForEvent{T}(PageEvent, WaitForEventOptions{T})"/>
         public Task<T> WaitForEvent<T>(PageEvent e, WaitForEventOptions<T> options = null)
+        {
+            ValidateArgumentsTypes();
+            var info = _pageEventsMap[e];
+            var eventTsc = new TaskCompletionSource<T>();
+            void PageEventHandler(object sender, T e)
+            {
+                if (options?.Predicate == null || options.Predicate(e))

```

Comment:
```suggestion
                if (options?.Predicate?.Invoke(e) == true)

```

Reply:
not sure this is equivalent to what we want. if `options.Predicate` is not given then we want to enter the if statement

Reply:
Good point :+1: